### ISSUE TITLE
Feature request: add option to delete attachments from matched messages, without folder download if possible

### ISSUE THREAD

**Issue by Pegasus-RPG** [RealRaven2000/FiltaQuilla#51]
I often need to mass-delete attachments from sent messages for my clients. Currently I have to mass-detach them to a file system folder, then delete them from there. It would be great if I could just tell FQ to delete the attachments directly.

Also, would this be possible if IMAP folder sync/download was disabled? That would be perfect!

**Comment by RealRaven2000** [comment]
> Also, would this be possible if IMAP folder sync/download was disabled? That would be perfect!
No. not as far as I know. I believe attachments are parts of the body of the email, there is no information in the header about them. That's what makes them so hard to work with, and literally excludes any meaningful filter conditions from working with attachemnts: sadly the filter testing still runs synchronously which would lead to catastrophic slowdowns in folders with many attachments.

By the way, this stuff is very hard to test. Might be easier to do a quick remote debugging session and see it crash on your computer than me trying to replicate your environment. Just a thought.

**Comment by Pegasus-RPG** [comment]
> I believe attachments are parts of the body of the email, there is no information in the header about them.

Sure, but could FQ's attachment handing possibly operate on each message as it is downloaded instead of having to have the entire folder pre-downloaded?

**Comment by RealRaven2000** [comment]
> > I believe attachments are parts of the body of the email, there is no information in the header about them.
> 
> Sure, but could FQ's attachment handing possibly operate on each message as it is downloaded instead of having to have the entire folder pre-downloaded?

possibly, who knows? I think I tried that and failed... It would be awesome if there was an IMAP command for deleting attachments. Maybe there is?

**Comment by RealRaven2000** [comment]
Probably setting to WONTFIX for now, I am using the internal detach function of Thunderbird and it forces saving the file. Just set the target folder to your bin and auto-schedule a cleanup
